_posts/2020-05-04-strangler-pattern.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title: Strangler Pattern
+date: 2020-05-04
+tags: architecture patterns
+author: Tim Myerscough
+image: img/strangler-pattern/Strangler_fig_boulder_katandra.jpg
+---
+
+<center><a href="https://commons.wikimedia.org/wiki/File:Strangler_fig_boulder_katandra.jpg"> <img alt="strangler fig" src="/img/strangler-pattern/Strangler_fig_boulder_katandra.jpg" /></a></center>
+<br/>
+
+Organisations who face replacing legacy applications often face an impossible choice.  Do you pay the high cost of a big-bang rewrite with the inherent disruption of a cutover, or do you keep the legacy application on life support?  
+
+Is there a better way?  One where you can break apart your legacy monolith.  Whilst moving to the cloud. And adopting Cloud Native practices.  Maybe there is...
+
+In the modern world of agile development, architecture has no end state.  Business must be able to respond to ever changing demands meaning software architecture must be equally adaptive.  A modern architecture supports small incremental changes with sufficient feedback loops to evaluate the effectiveness.  There are a number of approaches available when dealing with your legacy architecture and moving towards an evolutionary architecture, but one approach I'd like to cover in this post is the [Strangler Pattern](https://martinfowler.com/bliki/StranglerFigApplication.html).  I shall give an overview of the strangler pattern and also detail a couple of case studies of where we have applied it successfully.
+
+# The Context
+Consider migration to the cloud, [Infrastructure Modernisation](https://www.mechanicalrock.io/our-expertise/infrastructure-modernisation), or [Serverless technology](https://www.mechanicalrock.io/our-expertise/cloud-native-solutions).  How can we move faster whilst dealing with legacy systems?
```

Comment:
Okay I'm considering.... wtf do you want me to consider?

How about : "Infrastructure modernisation offers tangible benefits including reduced operating costs, scalable performance and greater uptime by using technologies such as serverless. Migrating legacy systems to the cloud however is not straight forward. How can we .. etc"

Reply:
general considering required.  Whatever you like.

But begrudgingly your version is also good.